src/lib.rs

```diff
@@ -508,6 +548,35 @@ fn reconstruct_cigar_from_segments_with_aligner(
     cigar_ops_to_cigar_string(&cigar_ops)
 }
 
+/// Reconstruct fastga CIGAR string from tracepoint segments using provided aligner
+///
+/// Takes an aligner parameter to allow callers to prepare an aligner and reuse it multiple times.
+fn reconstruct_fastga_cigar_from_segments_with_aligner(
+    segments: &[(usize, usize)],
+    a_seq: &[u8],
+    b_seq: &[u8],
+    a_start: usize,
+    b_start: usize,
+    aligner: &mut AffineWavefronts,
+) -> String {
+    let mut cigar_ops = Vec::new();
+    let mut current_a = a_start;
+    let mut current_b = b_start;
+
+    for &(_, b_len) in segments {
```

Comment:
Since fastga run the alignment regardless the chunk is pure insertion / deetion, I did the same thing over here.